src/Microsoft.Diagnostics.TestHelpers/TestConfiguration.cs

```diff
@@ -171,51 +171,146 @@ string GetInitialWorkingDir()
+                return isEquals ? leftValue == rightValue : leftValue != rightValue;
+            }
+            return true;
+        }
+
+        private bool TryGetParametersForFunction(string expression, string targetFunctionName, out List<string> exprParams)
+        {
+            int functionKeyworkIndex = expression.IndexOf($"{targetFunctionName}(");
+            if (functionKeyworkIndex == -1) {
+                exprParams = null;
+                return false;
+            }
 
-                    if (parts.Length == 2)
+            if (functionKeyworkIndex != 0 || functionKeyworkIndex >= 1 && expression[0] != '!' || !expression.EndsWith(')'))
+            {
+                throw new InvalidDataException($"Condition {expression} malformed. Currently only single-function conditions are supported.");
+            }
+
+            exprParams = new List<string>();
+            bool isWithinString = false;
+            bool expectDelimiter = false;
+            int curParsingIndex = functionKeyworkIndex + targetFunctionName.Length + 1;
+            StringBuilder resolvedValue = new StringBuilder();
+
+            // Account for the trailing parenthesis.
+            while(curParsingIndex + 1 < expression.Length)
+            {
+                char currentChar = expression[curParsingIndex];
+                // toggle string nesting on ', except if scaped
+                if (currentChar == '\'' && !(curParsingIndex > 0 && expression[curParsingIndex - 1] != '\\'))
+                {
+                    if (isWithinString)
                     {
-                        equal = true;
+                        exprParams.Add(resolvedValue.ToString());
+                        expectDelimiter = true;
                     }
-                    else
+
+                    isWithinString = !isWithinString;
+                }
+                else if (isWithinString)
+                {
+                    resolvedValue.Append(currentChar);
+                }
+                else if (currentChar == ',')
+                {
+                    if (!expectDelimiter)
                     {
-                        parts = conditionText.Split("!=");
-                        Assert.Equal(2, parts.Length);
-                        equal = false;
+                        throw new InvalidDataException($"Unexpected comma found within {expression}");
                     }
-                    // Resolve any config values in the condition
-                    string leftValue = ResolveProperties(config, parts[0]).Trim();
-                    string rightValue = ResolveProperties(config, parts[1]).Trim();
-
-                    // Now do the simple string comparison of the left/right sides of the condition
-                    return equal ? leftValue == rightValue : leftValue != rightValue;
+                    expectDelimiter = false;
+                }
+                else if (!Char.IsWhiteSpace(currentChar))
```

Comment:
The SOS test's use of Exists is failing here:

      <Option Condition="Exists('$(RepoRootDir)/.dotnet/shared/Microsoft.NETCore.App/$(RuntimeVersion11)')">